docs/plugins/hmr.md

```diff
@@ -145,6 +124,30 @@ if (environment.hmr) {
 }
 ```
 
+### Update src/app/app.module.ts to manage the state in HMR lifecycle:
+
+```ts
+import { StateOperations } from '@ngxs/store';
+import { NgxsHmrLifeCycle, NgxsStoreSnapshot } from '@ngxs/hmr-plugin';
+
+@NgModule({ .. })
+export class AppBrowserModule implements NgxsHmrLifeCycle<NgxsStoreSnapshot> {
+
+  public hmrNgxsStoreOnInit(ctx: StateOperations<NgxsStoreSnapshot>, snapshot: NgxsStoreSnapshot) {
```

Comment:
```suggestion
  public hmrNgxsStoreOnInit(ctx: StateContext<NgxsStoreSnapshot>, snapshot: NgxsStoreSnapshot) {
```
Could we rather use a `StateContext` as opposed to a `StateOperations`?
`StateContext` is already public, has a `patchState` method and would already be familiar to the users too.
I would prefer that `StateOperations` does not become part of the public API because its form is likely to change in the future.

Reply:
Done in latest commits to this branch.